<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Finding Pathways from PTMs: An Analysis Guide • PTMsToPathways</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Finding Pathways from PTMs: An Analysis Guide">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PTMsToPathways</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/PTMsToPathways.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Finding Pathways from PTMs: An Analysis Guide</h1>
                        <h4 data-toc-skip class="author">Nagashree
Avabhrath, Mikhail Bailey, Mark Grimes, Madison Moffett, Grant
Smith</h4>
            
      

      <div class="d-none name"><code>PTMsToPathways.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="this-package">This Package<a class="anchor" aria-label="anchor" href="#this-package"></a>
</h2>
<p>The PTMsToPathways package takes mass-spectrometry data of
post-translational modifications under different experimental conditions
and implicates pathways that are involved. These pathways are generated
based on analysis of which ptms cluster together (based on the same
environmental conditions) compared to how those proteins are known to
interact.</p>
<p><strong>An important note about this package:</strong> there are no
returned outputs from any of the functions. All outputs listed are
assigned to the Global Namespace in order to prevent loss of data and
promote ease of use.</p>
</div>
<div class="section level2">
<h2 id="this-file">This File<a class="anchor" aria-label="anchor" href="#this-file"></a>
</h2>
<p>This vignette is intended to be a step-by-step guide to walk users
through the process of using the PTMsToPathways package. It includes an
example pipeline demonstrating how to run the full analysis along with
descriptions of each function. this pipeline must be run in order as
subsequent steps require the data produced in previous steps. Estimated
run-times are included with each description and are based on a
preliminary dataset of ~9,000 post-translational modifications and 70
experimental conditions processed with a 12th Gen i5 processor and 8GB
of RAM.</p>
</div>
<div class="section level2">
<h2 id="starting-data">Starting Data<a class="anchor" aria-label="anchor" href="#starting-data"></a>
</h2>
<p>Below is an example of the input dataset:</p>
<pre><code><span><span class="co">#&gt;    Protein Mod Amino Acid H3122SEPTM.C1 H3122SEPTM.C2 H3122SEPTM.C3</span></span>
<span><span class="co">#&gt; 38    CTTN ack       K107      19.23272      19.85052      19.58630</span></span>
<span><span class="co">#&gt; 39    CTTN ack       K124      17.50665      19.73125      19.12403</span></span>
<span><span class="co">#&gt; 40    CTTN ack       K161            NA            NA            NA</span></span>
<span><span class="co">#&gt; 41    CTTN ack       K171            NA            NA            NA</span></span>
<span><span class="co">#&gt; 42    CTTN ack       K198      20.53742      20.89560      20.04847</span></span>
<span><span class="co">#&gt; 43    CTTN ack       K235      20.62203      20.74460      21.06424</span></span>
<span><span class="co">#&gt; 44    CTTN ack       K272            NA            NA            NA</span></span>
<span><span class="co">#&gt; 45    CTTN ack       K309      21.52164      21.03127      21.14289</span></span>
<span><span class="co">#&gt; 46    CTTN ack        K87      19.61967      20.63351      20.49436</span></span>
<span><span class="co">#&gt; 47    CTTN   p       S156      21.09816      23.48837      22.33529</span></span>
<span><span class="co">#&gt; 48    CTTN   p       S405            NA            NA            NA</span></span>
<span><span class="co">#&gt; 49    CTTN   p       S417            NA      18.22809      18.97026</span></span>
<span><span class="co">#&gt; 50    CTTN   p       S418            NA      18.22809      18.97026</span></span></code></pre>
<p>Note that the rowname are not important and will not be used. The
first column is the gene name associated with the protein. The second is
the type of post-translational modification that occurred. The third
column is the location of the post-translational modification. All of
subsequent columns, regardless of how many there are, are the
environmental conditions under which the post-translational
modifications occurred. NAs – which are importantly not zeroes –
represent condition-PTM combinations that were not studied while the
numeric values are data output by the mass spectrometer.</p>
</div>
<div class="section level2">
<h2 id="pipeline">Pipeline<a class="anchor" aria-label="anchor" href="#pipeline"></a>
</h2>
<div class="section level3">
<h3 id="step-1-make-cluster-list">Step 1: Make Cluster List<a class="anchor" aria-label="anchor" href="#step-1-make-cluster-list"></a>
</h3>
<div class="section level5">
<h5 id="code">Code<a class="anchor" aria-label="anchor" href="#code"></a>
</h5>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MakeClusterList.html">MakeClusterList</a></span><span class="op">(</span><span class="va">ptmtable</span>, name.columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, keeplength <span class="op">=</span> <span class="fl">2</span>, correlation.matrix.name <span class="op">=</span> <span class="st">"ptm.correlation.matrix"</span>, clusters.list.name <span class="op">=</span> <span class="st">"clusters.list"</span>, tsne.coords.name <span class="op">=</span> <span class="st">"all.tsne.coords"</span>, common.clusters.name <span class="op">=</span> <span class="st">"common.clusters"</span>, toolong <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/EU_MCL.png"><br><strong>Figure 1</strong> Example plot produced by MakeClusterList
calculated using Euclidean Distance. The Euclidean Distance between
every PTM is calculated and that information is put into a large matrix.
This matrix is then condensed using t-SNE to get coordinates in
two-dimensional space, which is what our figure shows. These data points
– and the PTMs they represent – are put into clusters based on these
positions. This process is also undergone using Spearman Dissimilarity
and the average of Euclidean Distance and Spearman Dissimilarity.</p>
<p>
 
</p>
<p><img src="vig_figs/MCL-output.png"><br><strong>Figure 2</strong> Output of MakeClusterList. This data is read
as two rows: the top row represents the cluster number. This number
isn’t important in and of itself, it just differentiates between the
different clusters. The second row represents the size, or how many PTMs
are in that cluster.</p>
<p>
 
</p>
<pre><code><span><span class="co">#&gt; $`1`</span></span>
<span><span class="co">#&gt;                                             PTM.Name group</span></span>
<span><span class="co">#&gt; 1                                      AARS ubi K747     1</span></span>
<span><span class="co">#&gt; 2                                    ANAPC5 ubi K289     1</span></span>
<span><span class="co">#&gt; 3                                      CUTA ubi K112     1</span></span>
<span><span class="co">#&gt; 4                                     CYHR1 ubi K349     1</span></span>
<span><span class="co">#&gt; 5                                      EEF2 ack K498     1</span></span>
<span><span class="co">#&gt; 6                                       F11R ubi K97     1</span></span>
<span><span class="co">#&gt; 7                                        GMPS ack K9     1</span></span>
<span><span class="co">#&gt; 8                                      HERC2 ubi K20     1</span></span>
<span><span class="co">#&gt; 9                                     KPNB1 ubi K541     1</span></span>
<span><span class="co">#&gt; 10                                     LASP1 ubi K59     1</span></span>
<span><span class="co">#&gt; 11                                     LNPEP ubi K32     1</span></span>
<span><span class="co">#&gt; 12                                    MRPS27 ubi K94     1</span></span>
<span><span class="co">#&gt; 13                                      NME1 ack K39     1</span></span>
<span><span class="co">#&gt; 14                                     PCNA ubi K110     1</span></span>
<span><span class="co">#&gt; 15                                      PKM ack K141     1</span></span>
<span><span class="co">#&gt; 16                                     PLAU ubi K403     1</span></span>
<span><span class="co">#&gt; 17                                     PLK1 ubi K492     1</span></span>
<span><span class="co">#&gt; 18                                   PNPLA2 ubi K435     1</span></span>
<span><span class="co">#&gt; 19                                    PSMC1 ubi K237     1</span></span>
<span><span class="co">#&gt; 20                                    RBCK1 ubi K342     1</span></span>
<span><span class="co">#&gt; 21                                    RPS15A ubi K12     1</span></span>
<span><span class="co">#&gt; 22                                    TCAF1 ubi K817     1</span></span>
<span><span class="co">#&gt; 23 TUBB4B ubi K379; TUBB2A ubi K379; TUBB2B ubi K379     1</span></span>
<span><span class="co">#&gt; 24                                    UIMC1 ubi K245     1</span></span>
<span><span class="co">#&gt; 25                                     USP5 ubi K318     1</span></span>
<span><span class="co">#&gt; 26                                    VAMP7 ubi K125     1</span></span>
<span><span class="co">#&gt; 27                                      VCP ubi K295     1</span></span></code></pre>
<p><strong>Figure 3</strong> First cluster created by Euclidean
Distance. The left column is the name of the PTM in this cluster. The
right column is the cluster number.</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h5>
<p>Make Cluster List is the first step in the analyzing one’s data. This
function takes the post-translational modification table and runs it
through three calculations of distance: Euclidean Distance, Spearman
Dissimilarity (1 - |Spearman Correlation|), and the average of the two
of these. These calculations find the ‘distance’ between ptms based upon
under what conditions they occur. In other words, they found how
dissimilar each pair of PTMs are. These matricies are then run through
t-SNE in order to put them into a 3-dimensional space. Please note:
t-SNE involves an element of randomness; in order to get the same
results, set.seed(#) must be called. A correlation table is also
produced based on the Spearman Correlation table.</p>
</div>
<div class="section level5">
<h5 id="input">Input<a class="anchor" aria-label="anchor" href="#input"></a>
</h5>
<ul>
<li>ptmtable
<ul>
<li>A data frame of post-translational modifications. Formatted with
numbered rows and the first column containing PTM names. The rest of the
column names should be drugs (experimental conditions). Values of the
table are numeric values that represent how much the PTM has reacted to
the drug or NA if that PTM wasn’t studied under that condition.<br>
</li>
</ul>
</li>
<li>name.columns
<ul>
<li>Numeric vector of the column number(s) containing the name or name
components of the post-translational modification; defaults to 1:3</li>
</ul>
</li>
<li>keeplength
<ul>
<li>MakeClusterList only saves subsets whose size is strictly
<em>greater</em> than keeplength; defaults to 2</li>
<li>example: [‘AARS’, ‘ABR’] will be discarded unless keeplength &lt;
2</li>
</ul>
</li>
<li>correlation.matrix.name
<ul>
<li>Desired name for the correlation matrix to be saved as; defaults to
ptm.correlation.matrix<br>
</li>
</ul>
</li>
<li>clusters.list.name
<ul>
<li>Desired name for the lists of clusters to be saved as; defaults to
clusters.list<br>
</li>
</ul>
</li>
<li><h3 id="tsne.coords.name">tsne.coords.name</h3></li>
<li><h3 id="common.clusters.name">common.clusters.name</h3></li>
<li>toolong
<ul>
<li>Threshold for cluster separation; defaults to 3.5</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h5>
<ul>
<li>ptm.correlation.matrix (or otherwise named by
correlation.matrix.name)
<ul>
<li>A data frame showing the correlation between ptms (as the rows and
the columns). NAs are placed along the diagonal so as to not skew the
data with self-similarity<br>
</li>
</ul>
</li>
<li>clusters.list (or otherwise named by list.name)
<ul>
<li>A list of three-dimensional data frames used to represent ptms in
space to show relationships between them based on distances. Based on
Euclidean Distance, Spearman Dissimilarity, and SED (the average between
the two)<br>
</li>
</ul>
</li>
<li><h3 id="all.tsne.coords-or-otherwise-named-by-tsne.coords.name">all.tsne.coords
(or otherwise named by tsne.coords.name)</h3></li>
<li><h3 id="common.clusters-or-otherwise-named-by-common.clusters.name">common.clusters
(or otherwise named by common.clusters.name)</h3></li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time"></a>
</h5>
<p>1hr 30mins</p>
</div>
</div>
<div class="section level3">
<h3 id="step-2-make-correlation-network">Step 2: Make Correlation Network<a class="anchor" aria-label="anchor" href="#step-2-make-correlation-network"></a>
</h3>
<div class="section level5">
<h5 id="code-1">Code<a class="anchor" aria-label="anchor" href="#code-1"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MakeCorrelationNetwork.html">MakeCorrelationNetwork</a></span><span class="op">(</span><span class="va">common.clusters</span>, <span class="va">ptm.correlation.matrix</span>, ptm.cccn.name <span class="op">=</span> <span class="st">"ptm.cccn"</span>, gene.cccn.name <span class="op">=</span> <span class="st">"gene.cccn"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/cccn_matrix.png"><br><strong>Figure 4</strong> First 17 rows and columns of the cccn_matrix
produced by MakeCorrelationNetwork. PTMs that cluster together in all
three distance metrics have entries that represent how strongly they
correlate, or how alike their responses are under the same environment.
PTMs that don’t cluster in all three distance metrics are correlated by
a 0. Self-correlations are also marked by a 0 to prevent self-similarity
skewing.</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description-1">Description<a class="anchor" aria-label="anchor" href="#description-1"></a>
</h5>
<p>Make Correlation Network first finds the intersection between the
Euclidian, Spearman, and SED cluster matrices in order to find the
intersection between the three groups. It then adds the Genes in these
PTMs to a list of common clusters and turns it into an adjacency matrix
with two variants, one at the PTM level and one at the gene level. This
adjacency matrix is used to filter relevant data — clusters — from the
Spearman correlation matrix. The resultant cocluster correlation network
shows strength of relationships between proteins using the common
clusters between the three distance metrics.</p>
</div>
<div class="section level5">
<h5 id="input-1">Input<a class="anchor" aria-label="anchor" href="#input-1"></a>
</h5>
<ul>
<li>clusterlist
<ul>
<li>A list of three-dimensional data frames used to represent ptms in
space to show relationships between them based on distances. Based on
Euclidean Distance, Spearman Dissimilarity, and SED (the average between
the two)<br>
</li>
</ul>
</li>
<li>ptm.correlation.matrix
<ul>
<li>A data frame showing the correlation between ptms (as the rows and
the columns). NAs are placed along the diagonal so as to not skew the
data with self-similarity.<br>
</li>
</ul>
</li>
<li>clusters.name
<ul>
<li>Desired name for the common clusters output; defaults to
common.clusters</li>
</ul>
</li>
<li>filtered.ptm.cor.name
<ul>
<li>Desired name for the correlation matrix that contains only PTMs
found in common clusters</li>
</ul>
</li>
<li>cccn.name
<ul>
<li>Desired name for the cocluster correlation network matrix; defaults
to cccn.matrix</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-1">Output<a class="anchor" aria-label="anchor" href="#output-1"></a>
</h5>
<ul>
<li>common.clusters (or otherwise named by common.clusters.name)
<ul>
<li>The list of common clusters between all three distance metrics
(Euclidean, Spearman, and SED)<br>
</li>
</ul>
</li>
<li>cccn.matrix (or otherwise named by cccn.name)
<ul>
<li>A matrix showing strength of relationships between proteins using
common clusters between the three distance metrics</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-1">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-1"></a>
</h5>
<p>2hrs 30mins</p>
</div>
</div>
<div class="section level3">
<h3 id="step-3-retrieve-database-edgefiles">Step 3: Retrieve Database Edgefiles<a class="anchor" aria-label="anchor" href="#step-3-retrieve-database-edgefiles"></a>
</h3>
<div class="section level4">
<h4 id="description-2">Description<a class="anchor" aria-label="anchor" href="#description-2"></a>
</h4>
<p>PPI (protein-protein interaction) databases are consulted in order to
filter the clusters by proteins that are known to interact with each
other as well as how strongly they are known to interact. The standard
PPI database that is used is STRINGdb, and getting data from this
database is the first step. This is accomplished with the function
GetSTRINGdb. Please note, however, that the user may consult any
database that they choose. After getting STRINGdb data (or not), the
user runs MakeDBInput which produces a text file of all of their gene
names. This information can be copy and pasted into any database that
the user chooses in order to get other PPI networks. Step three is
getting a GeneMANIA network, which is also optional but recommended. The
user pastes their input data into GeneMANIA on the Cytoscape app and
saves the edgefile and the nodetable. These files are then input into
ProcessGMEdgefile in order to sort the data.</p>
<p>Note again that the database input can be used in any PPI database
that the user chooses, though this package only explicitly supports
STRINGdb and GeneMANIA. If another database is chosen, its file will
have to be filtered manually by the user before moving on to step 4. The
file should have three columns. Column one and two should strictly be
labeled “Gene.1” and “Gene.2” in order to integrate with other PPI
databases. The third column should contain the edgeweight and may be
named however the user chooses. It is recommended, though, that the
database is specified as well as the term ‘weight’ in the column
name.</p>
</div>
<div class="section level4">
<h4 id="part-1-get-stringdb-data">Part 1 — Get STRINGdb Data<a class="anchor" aria-label="anchor" href="#part-1-get-stringdb-data"></a>
</h4>
<div class="section level5">
<h5 id="code-2">Code<a class="anchor" aria-label="anchor" href="#code-2"></a>
</h5>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GetSTRINGdb.html">GetSTRINGdb</a></span><span class="op">(</span><span class="va">gene.cccn</span>, stringdb.name <span class="op">=</span> <span class="st">"stringdb.edges"</span>, nodenames.name <span class="op">=</span> <span class="st">"nodenames"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/string-edges.png"><br><strong>Figure 5</strong> First 18 rows of stringdb.edges produced by
GetSTRINGdb</p>
<p>
 
</p>
<p><img src="vig_figs/nodenames.png"><br><strong>Figure 6</strong> First 18 rows of nodenames produced by
GetSTRINGdb</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="input-2">Input<a class="anchor" aria-label="anchor" href="#input-2"></a>
</h5>
<ul>
<li>cccn.matrix
<ul>
<li>A matrix showing strength of relationships between proteins using
the common clusters between the three distance metrics<br>
</li>
</ul>
</li>
<li>stringdb.name
<ul>
<li>Desired name for STRINGdb data; defaults to stringdb.edges<br>
</li>
</ul>
</li>
<li>nodenames.name
<ul>
<li>Desired name for list of gene names; defaults to nodenames</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-2">Output<a class="anchor" aria-label="anchor" href="#output-2"></a>
</h5>
<ul>
<li>stringdb.edges (or otherwise named by STRINGdb.name)
<ul>
<li>Data frame of consisting of the network of interactions from the
genes of study pulled from the STRINGdb database<br>
</li>
</ul>
</li>
<li>nodenames (or otherwise named by nodenames.name)
<ul>
<li>List of gene names</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-2">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-2"></a>
</h5>
<p>30 mins</p>
</div>
</div>
<div class="section level4">
<h4 id="part-2-get-file-for-database-input">Part 2 — Get File for Database Input<a class="anchor" aria-label="anchor" href="#part-2-get-file-for-database-input"></a>
</h4>
<div class="section level5">
<h5 id="code-3">Code<a class="anchor" aria-label="anchor" href="#code-3"></a>
</h5>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MakeDBInput.html">MakeDBInput</a></span><span class="op">(</span><span class="va">gene.cccn</span>, file.path.name <span class="op">=</span> <span class="st">"db_nodes.txt"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/db_nodes-txt.png"></p>
<p><strong>Figure 7</strong> First 15 lines from the produced text
file</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="input-3">Input<a class="anchor" aria-label="anchor" href="#input-3"></a>
</h5>
<ul>
<li>cccn.matrix
<ul>
<li>A matrix showing strength of relationships between proteins using
the common clusters between the three distance metrics (Euclidean,
Spearman, and Combined (SED))<br>
</li>
</ul>
</li>
<li>file.path.name
<ul>
<li>Desired path for file to be saved as; defaults to db_nodes.txt</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-3">Output<a class="anchor" aria-label="anchor" href="#output-3"></a>
</h5>
<ul>
<li>db_nodes.txt (or otherwise named by file.path.name)
<ul>
<li>A text file with the names of all of the genes<br>
</li>
<li>Saved into the user’s working directory unless otherwise stated in
file.path.name</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-3">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-3"></a>
</h5>
<p>1s</p>
</div>
</div>
<div class="section level4">
<h4 id="part-3-process-genemania-file">Part 3 — Process GeneMANIA File<a class="anchor" aria-label="anchor" href="#part-3-process-genemania-file"></a>
</h4>
<div class="section level5">
<h5 id="code-4">Code<a class="anchor" aria-label="anchor" href="#code-4"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ProcessGMEdgefile.html">ProcessGMEdgefile</a></span><span class="op">(</span><span class="va">gm.edgefile.path</span>, <span class="va">gm.nodetable.path</span>, <span class="va">db_nodes.path</span>, gm.network.name <span class="op">=</span> <span class="st">"gm.network"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/gm_network.png"></p>
<p><strong>Figure 8</strong> First 44 rows of the GeneMANIA network</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="input-4">Input<a class="anchor" aria-label="anchor" href="#input-4"></a>
</h5>
<ul>
<li>gm.edgefile.path
<ul>
<li>Path to the GeneMANIA edgefile</li>
</ul>
</li>
<li>gm.nodetable.path
<ul>
<li>Path to the GeneMANIA nodetable</li>
</ul>
</li>
<li>db_nodes.path
<ul>
<li>Path to the nodenames file created in make_db_input</li>
</ul>
</li>
<li>gm.network.name
<ul>
<li>Desired name for the output of the GeneMANIA network, defaults to
gm.network</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-4">Output<a class="anchor" aria-label="anchor" href="#output-4"></a>
</h5>
<ul>
<li>gm.network (or otherwise named by gm.network.name)
<ul>
<li>Data frame of consisting of the network of interactions from the
genes of study</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-4">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-4"></a>
</h5>
<p>10s</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="step-4-build-ppi-network">Step 4: Build PPI Network<a class="anchor" aria-label="anchor" href="#step-4-build-ppi-network"></a>
</h3>
<div class="section level5">
<h5 id="code-5">Code<a class="anchor" aria-label="anchor" href="#code-5"></a>
</h5>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/BuildPPINetwork.html">BuildPPINetwork</a></span><span class="op">(</span>stringdb.edges <span class="op">=</span> <span class="cn">NA</span>, gm.network <span class="op">=</span> <span class="cn">NA</span>, db.filepaths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>, ppi.network.name <span class="op">=</span> <span class="st">"ppi.network"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/ppi_network.png"><br><strong>Figure 9</strong> First 19 rows of the ppi_network produced by
find_ppi_edges</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description-3">Description<a class="anchor" aria-label="anchor" href="#description-3"></a>
</h5>
<p><strong>Note:</strong> Examples take about 5-10 minutes to run.</p>
<p>The purpose of BuildPPINetwork is to combine the data about desired
protein-protein interactions from each of the provided databases. This
package explicitly processes STRINGdb and GeneMANIA in Step 3, but the
user may consult as many databases as desired (the processing for which
is described in Step 3). BuildPPINetwork combines these databases as
efficiently as possible while retaining desired edgeweights from each
database. BuildPPINetwork also removes duplicate rows from the data
frame.</p>
</div>
<div class="section level5">
<h5 id="input-5">Input<a class="anchor" aria-label="anchor" href="#input-5"></a>
</h5>
<ul>
<li>stringdb.edges (produced optionally in step 3)
<ul>
<li>Data frame of consisting of the network of interactions from the
genes of study pulled from the STRINGdb database; defaults to NA<br>
</li>
</ul>
</li>
<li>gm.network (produced optionally in step 3)
<ul>
<li>GeneMANIA network of protein-protein interactions from the genes of
study; defaults to NA<br>
</li>
</ul>
</li>
<li>db.filepaths (produced optionally in step 3)
<ul>
<li>A vector of paths to the additional ppi network files; defaults to
an empty vector<br>
</li>
</ul>
</li>
<li>ppi.network.name
<ul>
<li>Desired name for the output protein-protein interaction network
using all entered database input; defaults to ppi.network</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-5">Output<a class="anchor" aria-label="anchor" href="#output-5"></a>
</h5>
<ul>
<li>ppi.network (or otherwise named by ppi.network.name)
<ul>
<li>A dataframe representing how strongly proteins are known to interact
based on previous research contained within provided database
networks</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-5">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-5"></a>
</h5>
<p>1hr (only using STRINGdb and GeneMANIA)</p>
</div>
</div>
<div class="section level3">
<h3 id="step-5-cluster-filtered-network">Step 5: Cluster Filtered Network<a class="anchor" aria-label="anchor" href="#step-5-cluster-filtered-network"></a>
</h3>
<div class="section level5">
<h5 id="code-6">Code<a class="anchor" aria-label="anchor" href="#code-6"></a>
</h5>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ClusterFilteredNetwork.html">ClusterFilteredNetwork</a></span><span class="op">(</span><span class="va">gene.cccn</span>, <span class="va">ppi.network</span>, cfn.name <span class="op">=</span> <span class="st">"cfn"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/cfn.png"><br><strong>Figure 10</strong> First 19 rows of the cfn produced by
ClusterFilteredNetwork</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description-4">Description<a class="anchor" aria-label="anchor" href="#description-4"></a>
</h5>
<p>Cluster Filtered Network checks all of the edges in the PPI network
to see ensure that both of the genes are within our cocluster
correlation network and that its weight is nonzero. If either of these
conditions are not met, then it will be removed from the list of PPI
edges. This new, cluster filtered network is then returned.</p>
</div>
<div class="section level5">
<h5 id="input-6">Input<a class="anchor" aria-label="anchor" href="#input-6"></a>
</h5>
<ul>
<li>cccn.matrix
<ul>
<li>A matrix showing strength of relationships between proteins using
the common clusters between the three distance metrics<br>
</li>
</ul>
</li>
<li>ppi.network
<ul>
<li>A dataframe representing how strongly proteins are known to
interact<br>
</li>
</ul>
</li>
<li>cfn.name
<ul>
<li>Desired name for the output of ClusterFilteredNetwork; defaults to
cfn</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-6">Output<a class="anchor" aria-label="anchor" href="#output-6"></a>
</h5>
<ul>
<li>cfn (or otherwise named by cfn.name)
<ul>
<li>A version of ppi.network with only the edges that exist in
cccn.matrix and have non-zero weights</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-6">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-6"></a>
</h5>
<p>10mins</p>
</div>
</div>
<div class="section level3">
<h3 id="step-6-pathway-crosstalk-network">Step 6: Pathway Crosstalk Network<a class="anchor" aria-label="anchor" href="#step-6-pathway-crosstalk-network"></a>
</h3>
<div class="section level5">
<h5 id="code-7">Code<a class="anchor" aria-label="anchor" href="#code-7"></a>
</h5>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/PathwayCrosstalkNetwork.html">PathwayCrosstalkNetwork</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"bioplanet.csv"</span>, <span class="va">common.clusters</span>, edgelist.name <span class="op">=</span> <span class="st">"edgelist"</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/PTP_edgelist.png"></p>
<p><strong>Figure 12</strong> Graph in Cytoscape - Didn’t add cuz not
good yet Will exist at some point</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description-5">Description<a class="anchor" aria-label="anchor" href="#description-5"></a>
</h5>
<p>Note: This step is directory sensitive. You can check and set your
directory in R using getwd() and setwd(“yourdirectoryhere”)
respectively. It needs a path to the bioplanet file and will put an
edgelist file in your working directory, or getwd(). If you cannot find
a file, please check your directories first.</p>
<p>Pathway Crosstalk Network is the final step in the pipeline. It
requires input of an external database, bioplanet, which consists of
groups of genes (proteins) involved in various cellular processes. The
PCN turns this database into a list of pathways and converts those
pathways into a pathway x pathway edgelist that possesses multiple
weights, a jaccard similarity and a score derrived from Cluster-Pathway
Evidence using common clusters found in Make Correlation Network.</p>
</div>
<div class="section level5">
<h5 id="input-7">Input<a class="anchor" aria-label="anchor" href="#input-7"></a>
</h5>
<ul>
<li>file
<ul>
<li>File path for the pathway data from Bioplanet (can also take in a
data frame that is formatted correctly like the one found in the data
directory, but this is not recommended as formatting must match sample
data exactly)<br>
</li>
</ul>
</li>
<li>common.clusters
<ul>
<li>The list of common clusters between all three distance metrics
(Euclidean, Spearman, and SED) made by MakeCorrelationNetwork<br>
</li>
</ul>
</li>
<li>edgelist.name
<ul>
<li>Intended for use in Cytoscape.<br>
</li>
<li>The desired name of the Pathway to Pathway edgelist file created
(‘.csv’ will automatically be added to the end for you); defaults to
edgelist</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-7">Output<a class="anchor" aria-label="anchor" href="#output-7"></a>
</h5>
<ul>
<li>edgelist (or otherwise named by edgelist.name)
<ul>
<li>An edgelist file that is created in the working directory. Contains
pathway source-target columns, with edge weights of their jaccard
similarity and their Cluster-Pathway Evidence score. Info about the
Cluster-Pathway Evidence score can be found at: <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010690" class="external-link uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010690</a>
</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-7">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-7"></a>
</h5>
<p>UNKNOWN</p>
</div>
</div>
<div class="section level3">
<h3 id="step-7-graph-cluster-filtered-network">Step 7: Graph Cluster Filtered Network<a class="anchor" aria-label="anchor" href="#step-7-graph-cluster-filtered-network"></a>
</h3>
<div class="section level5">
<h5 id="code-8">Code<a class="anchor" aria-label="anchor" href="#code-8"></a>
</h5>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GraphCfn.html">GraphCfn</a></span><span class="op">(</span><span class="va">cfn</span>, Network.title <span class="op">=</span> <span class="st">"cfn"</span>, Network.collection <span class="op">=</span> <span class="st">"PTMsToPathways"</span>,</span>
<span>                     background.color <span class="op">=</span> <span class="st">'#fcf3cf'</span>, edge.label.color <span class="op">=</span> <span class="st">'#17202a'</span>, edge.line.color <span class="op">=</span> <span class="st">'#abb2b9'</span>, node.border.color <span class="op">=</span> <span class="st">'#145a32'</span>,</span>
<span>                     node.label.color <span class="op">=</span> <span class="st">'#145a32'</span>, node.fill.color <span class="op">=</span> <span class="st">'#a9dfbf'</span>,</span>
<span>                     default.font <span class="op">=</span> <span class="st">"Times New Roman"</span>, node.font.size <span class="op">=</span> <span class="fl">12</span>, edge.font.size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                     edge.line.style <span class="op">=</span> <span class="st">'SOLID'</span>, source.arrow <span class="op">=</span> <span class="st">'NONE'</span>, target.arrow <span class="op">=</span> <span class="st">'NONE'</span>, node.shape <span class="op">=</span> <span class="st">"OCTAGON"</span>,</span>
<span>                     node.size <span class="op">=</span> <span class="fl">50</span>, edge.width <span class="op">=</span> <span class="fl">2</span>, border.width <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                     edge.opacity <span class="op">=</span> <span class="fl">175</span>, edge.label.opacity <span class="op">=</span> <span class="fl">255</span>, border.opacity <span class="op">=</span> <span class="fl">255</span>, node.label.opacity <span class="op">=</span> <span class="fl">255</span>, node.fill.opacity <span class="op">=</span> <span class="fl">255</span><span class="op">)</span></span></code></pre></div>
<center>
<p><img src="vig_figs/cytoscape_graph.png"><br><strong>Figure 13</strong> Default cytoscape graph</p>
</center>
<p>
 
</p>
</div>
<div class="section level5">
<h5 id="description-6">Description<a class="anchor" aria-label="anchor" href="#description-6"></a>
</h5>
<p>This optional function GraphCfn creates a graph using the cluster
filtered network in the Cytoscape app.</p>
</div>
<div class="section level5">
<h5 id="input-8">Input<a class="anchor" aria-label="anchor" href="#input-8"></a>
</h5>
<ul>
<li>cfn
<ul>
<li>A version of ppi.network with only the edges that exist in
cccn.matrix and have non-zero weights<br>
-Network.title<br>
</li>
<li>Desired title for the created Cytoscape Network<br>
</li>
</ul>
</li>
<li>Network.collection
<ul>
<li>Desired name for the collection created on Cytoscape in which the
network will reside</li>
</ul>
</li>
<li>background.color
<ul>
<li>Hex code of background color of graph; defaults to ‘#fcf3cf’<br>
</li>
</ul>
</li>
<li>edge.label.color
<ul>
<li>Hex code of edge label color of graph; defaults to ‘#17202a’<br>
</li>
</ul>
</li>
<li>edge.line.color
<ul>
<li>Hex code of edge line color of graph; defaults to ‘#abb2b9’<br>
</li>
</ul>
</li>
<li>node.border.color
<ul>
<li>Hex code of node border color of graph; defaults to ‘#145a32’<br>
</li>
</ul>
</li>
<li>node.label.color
<ul>
<li>Hex code of node label color of graph; defaults to ‘#145a32’<br>
</li>
</ul>
</li>
<li>node.fill.color
<ul>
<li>Hex code of node fill color of graph; defaults to ‘#a9dfbf’</li>
</ul>
</li>
<li>default.font
<ul>
<li>Font style of edge and node names; defaults to “Times New
Roman”<br>
</li>
</ul>
</li>
<li>node.font.size
<ul>
<li>Font size of the node name; defaults to 12<br>
</li>
</ul>
</li>
<li>edge.font.size
<ul>
<li>Font size of the edge name; defaults to 8</li>
</ul>
</li>
<li>edge.line.style
<ul>
<li>Type of edge style; defaults to “SOLID”<br>
</li>
<li>options include: “PARALLEL_LINES”, “MARQUEE_EQUAL”, “DOT”,
“EQUAL_DASH”, “LONG_DASH”, “CONTIGUOUS_ARROW”, “MARQUEE_DASH”,
“DASH_DOT”, “BACKWARD_SLASH”, “FORWARD_SLASH”, “VERTICAL_SLASH”,
“SOLID”, “SEPARATE_ARROW”, “MARQUEE_DASH_DOT”, “ZIGZAG”,
“SINEWAVE”<br>
</li>
</ul>
</li>
<li>source.arrow
<ul>
<li>Type of arrow coming from the source gene; defaults to “NONE”<br>
</li>
<li>options include: “DELTA”, “DIAMOND”, “OPEN_CIRCLE”, “CIRCLE”,
“OPEN_HALF_CIRCLE”, “CROSS_OPEN_DELTA”, “DELTA_SHORT_1”, “CROSS_DELTA”,
“OPEN_DELTA”, “OPEN_DIAMOND”, “DIAMOND_SHORT_1”, “DELTA_SHORT_2”,
“OPEN_SQUARE”, “NONE”, “SQUARE”, “DIAMOND_SHORT_2”, “T”, “HALF_BOTTOM”,
“HALF_TOP”, “ARROW_SHORT”, “HALF_CIRCLE”<br>
</li>
</ul>
</li>
<li>target.arrow
<ul>
<li>Type of arrow going to the target gene; defaults to “NONE”<br>
</li>
<li>options include: “DELTA”, “DIAMOND”, “OPEN_CIRCLE”, “CIRCLE”,
“OPEN_HALF_CIRCLE”, “CROSS_OPEN_DELTA”, “DELTA_SHORT_1”, “CROSS_DELTA”,
“OPEN_DELTA”, “OPEN_DIAMOND”, “DIAMOND_SHORT_1”, “DELTA_SHORT_2”,
“OPEN_SQUARE”, “NONE”, “SQUARE”, “DIAMOND_SHORT_2”, “T”, “HALF_BOTTOM”,
“HALF_TOP”, “ARROW_SHORT”, “HALF_CIRCLE”<br>
</li>
</ul>
</li>
<li>node.shape
<ul>
<li>Shape of the node; defaults to “OCTAGON”<br>
</li>
<li>options include: “RECTANGLE”, “VEE”, “OCTAGON”, “ELLIPSE”,
“PARALLELOGRAM”, “DIAMOND”, “HEXAGON”, “ROUND_RECTANGLE”,
“TRIANGLE”</li>
</ul>
</li>
<li>node.size
<ul>
<li>Size of the node; defaults to 50<br>
</li>
<li>PLEASE NOTE: width and height can be changed independently using
RCy3 directly
<ul>
<li>first run lockNodeDimensions(FALSE) and then setNodeWidthDefault()
and setNodeHeightDefault()<br>
</li>
</ul>
</li>
</ul>
</li>
<li>edge.width
<ul>
<li>Width of the edge line; defaults to 2<br>
</li>
</ul>
</li>
<li>border.width
<ul>
<li>Width of the border of the node; defaults to 1</li>
</ul>
</li>
<li>edge.opacity
<ul>
<li>Opacity of the edge line on a scale of 0 - 255 with 0 being
transparent; defaults to 175<br>
</li>
</ul>
</li>
<li>edge.label.opacity
<ul>
<li>Opacity of the edge label on a scale of 0 - 255 with 0 being
transparent; defaults to 255<br>
</li>
</ul>
</li>
<li>border.opacity
<ul>
<li>Opacity of the node border on a scale of 0 - 255 with 0 being
transparent; defaults to 255<br>
</li>
</ul>
</li>
<li>node.label.opacity
<ul>
<li>Opacity of the node label on a scale of 0 - 255 with 0 being
transparent; defaults to 255<br>
</li>
</ul>
</li>
<li>node.fill.opacity
<ul>
<li>Opacity of the node fill on a scale of 0 - 255 with 0 being
transparent; defaults to 255</li>
</ul>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="output-8">Output<a class="anchor" aria-label="anchor" href="#output-8"></a>
</h5>
<ul>
<li>No actual output or assignment<br>
</li>
<li>Creates a graph in the cytoscape interface</li>
</ul>
</div>
<div class="section level5">
<h5 id="estimated-run-time-8">Estimated run-time<a class="anchor" aria-label="anchor" href="#estimated-run-time-8"></a>
</h5>
<p>UNKNOWN</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Mark Grimes, Mikhail Bailey, Grant Smith, Madison Moffett, Nagashree Avabhrath.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
